controllers/release/release_controller_test.go

```diff
@@ -17,129 +17,74 @@ limitations under the License.
-	)
-
-	BeforeEach(func() {
-		releasePlan = &appstudiov1alpha1.ReleasePlan{
-			ObjectMeta: metav1.ObjectMeta{
-				GenerateName: "test-releaseplan-",
-				Namespace:    testNamespace,
-				Labels: map[string]string{
-					appstudiov1alpha1.AutoReleaseLabel: "true",
-				},
-			},
-			Spec: appstudiov1alpha1.ReleasePlanSpec{
-				Application: "test-app",
-				Target:      testNamespace,
-			},
-		}
-		Expect(k8sClient.Create(ctx, releasePlan)).Should(Succeed())
-
-		release = &appstudiov1alpha1.Release{
-			TypeMeta: metav1.TypeMeta{
-				APIVersion: testApiVersion,
-				Kind:       "Release",
-			},
-			ObjectMeta: metav1.ObjectMeta{
-				GenerateName: "test-release-",
-				Namespace:    testNamespace,
-			},
-			Spec: appstudiov1alpha1.ReleaseSpec{
-				Snapshot:    "test-snapshot",
-				ReleasePlan: releasePlan.GetName(),
-			},
-		}
-		Expect(k8sClient.Create(ctx, release)).Should(Succeed())
-
-		req = ctrl.Request{
-			NamespacedName: types.NamespacedName{
-				Namespace: release.Namespace,
-				Name:      release.Name,
-			},
-		}
-
-		webhookInstallOptions := &testEnv.WebhookInstallOptions
+var _ = Describe("Release Controller", Ordered, func() {
 
-		var err error
-		manager, err = ctrl.NewManager(cfg, ctrl.Options{
-			Scheme:             clientsetscheme.Scheme,
-			Host:               webhookInstallOptions.LocalServingHost,
-			Port:               webhookInstallOptions.LocalServingPort,
-			CertDir:            webhookInstallOptions.LocalServingCertDir,
-			MetricsBindAddress: "0", // this disables metrics
-			LeaderElection:     false,
+	Context("When NewReleaseReconciler is called", func() {
+		It("creates and return a new Reconciler", func() {
+			Expect(reflect.TypeOf(NewReleaseReconciler(k8sClient, &ctrl.Log, scheme.Scheme))).To(Equal(reflect.TypeOf(&Reconciler{})))
 		})
-		Expect(err).NotTo(HaveOccurred())
-		releaseReconciler = NewReleaseReconciler(k8sClient, &logf.Log, &scheme)
 	})
 
-	AfterEach(func() {
-		err := k8sClient.Delete(ctx, release)
-		Expect(err == nil || errors.IsNotFound(err)).To(BeTrue())
-		err = k8sClient.Delete(ctx, releasePlan)
-		Expect(err == nil || errors.IsNotFound(err)).To(BeTrue())
-	})
-
-	It("can create and return a new Reconciler object", func() {
-		Expect(reflect.TypeOf(releaseReconciler)).To(Equal(reflect.TypeOf(&Reconciler{})))
-	})
-
-	It("can fail when Reconcile fails to prepare the adapter when release is not found", func() {
-		Expect(k8sClient.Delete(ctx, release)).Should(Succeed())
-		Eventually(func() error {
-			_, err := releaseReconciler.Reconcile(ctx, req)
-			return err
-		}).Should(BeNil())
+	// For the Reconcile function test we don't want to make a successful call as it will call every single operation
+	// defined there. We don't have any control over the operations being executed, and we want to keep a clean env for
+	// the adapter tests.
+	Context("When Reconcile is called", func() {
+		It("should reconcile the release but do nothing", func() {
```

Comment:
I did not get what this mean. But the test says we want to know if the object type matches `reconcile.Result{}` and no `err`. Maybe the description could be re-elaborated.

Reply:
Changed to `should succeed even if the release is not found`.